runrole

```
@@ -19,21 +20,34 @@ if [ ! -f $PLAYBOOK ]; then
     exit 1
 fi
 
-if [ $# -eq 0 ] || [ "$2" == "--reinstall" ] || [ "$3" == "--reinstall" ]; then
+# https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash/14203146#14203146
+while [[ $# -gt 0 ]]; do
+    case $1 in    # if-else would do the job, but case statement allows for future changes...
+        --reinstall)
+            ARGS="$ARGS\"reinstall\":True,"    # Needs boolean not string so use JSON list
+            REINSTALL=true
+            ;;
+        *)
+            if [ ! -v ROLE_NAME ]; then    # Test whether var is not yet set
+                ROLE_NAME="$1"
+                export ANSIBLE_LOG_PATH="$CWD/iiab-debug.log"
+            else
+                export ANSIBLE_LOG_PATH="$1"
```

So this goes `./runrole ROLE_NAME ANSIBLE_LOG_PATH` where `ANSIBLE_LOG_PATH` is optional and `--reinstall` can be stuck in anywhere after `./runrole`?

If that's right, maybe you could assert (with a "too many arguments" failure) that `ANSIBLE_LOG_PATH` is not already set. The way it is now, you could have extra arguments that don't do anything and it only takes the last arg as `ANSIBLE_LOG_PATH`. It wouldn't break anything but maybe you could "fail fast" for the benefit of someone who accidentally put in an extra argument.

---

Done: only hard-core implementers use `./runrole` but I agree with the spirit...

- So now it will at least warn people who blindly try `./runrole <ROLE1> <ROLE2> <ROLE3>`

- Though I'm not going to worry about users who specify `--reinstall` multiple time (as you say, no harm done!)